collector/loadavg.go

```diff
@@ -3,18 +3,15 @@
 package collector
 
 import (
+	"errors"
 	"fmt"
-	"io/ioutil"
-	"strconv"
-	"strings"
 
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/log"
 )
 
-const (
-	procLoad = "/proc/loadavg"
-)
+// #include <stdlib.h>
```

Comment:
FreeBSD has deprecated procfs long ago and there is no other way to collect stats other than C APIs.
Here is an example of linked libraries on a FreeBSD (10.1) system:
```
	libkvm.so.6 => /lib/libkvm.so.6 (0x800eef000)
	libdevstat.so.7 => /lib/libdevstat.so.7 (0x8010f7000)
	libthr.so.3 => /lib/libthr.so.3 (0x8012fd000)
	libc.so.7 => /lib/libc.so.7 (0x801522000)
```

For loadavg I thought maybe it's better to use stdlib.h which is avalable on all/most *nixes. (I didn't know it will link to libc dynamically, maybe there is a way for static linking?)
But you are right regarding cross compilation.
I'll split the modules, one for linux and one for other platforms(assuming they have stdlib.h available).